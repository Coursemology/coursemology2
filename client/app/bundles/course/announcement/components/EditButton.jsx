import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  onClick: PropTypes.func.isRequired,
};

class EditButton extends React.Component {
  render() {
    return (
      <span className="btn btn-default edit" title="Edit Announcement" onClick={() => this.props.onClick()}>
        <i className="fa fa-edit" />
      </span>
    );
  }
}

EditButton.propTypes = propTypes;

export default EditButton;
