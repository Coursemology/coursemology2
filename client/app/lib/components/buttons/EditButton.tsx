import { IconButton, IconButtonProps } from '@mui/material';
import Edit from '@mui/icons-material/Edit';

interface Props extends IconButtonProps {
  onClick: (SyntheticEvent: any) => void;
}

const EditButton = ({ onClick, ...props }: Props): JSX.Element => (
  <IconButton onClick={onClick} color="inherit" {...props}>
    <Edit />
  </IconButton>
);

export default EditButton;
