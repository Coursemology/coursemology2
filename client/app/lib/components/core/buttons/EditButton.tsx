import { SyntheticEvent } from 'react';
import { IconButton, IconButtonProps, Tooltip } from '@mui/material';
import Edit from '@mui/icons-material/Edit';

interface Props extends IconButtonProps {
  onClick: (e: SyntheticEvent) => void;
  tooltip?: string;
}

const EditButton = ({
  onClick,
  tooltip = '',
  ...props
}: Props): JSX.Element => (
  <Tooltip title="Edit">
    <span>
      <IconButton
        onClick={onClick}
        color="inherit"
        {...props}
        data-testid="EditIconButton"
      >
        <Edit data-testid="EditIcon" />
      </IconButton>
    </span>
  </Tooltip>
);

export default EditButton;
