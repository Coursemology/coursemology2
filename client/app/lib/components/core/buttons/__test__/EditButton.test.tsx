import {
  render,
  RenderResult,
  screen,
  waitForElementToBeRemoved,
} from 'test-utils';

import EditButton from '../EditButton';

let documentBody: RenderResult;

describe('<EditButton />', () => {
  beforeEach(() => {
    documentBody = render(<EditButton onClick={jest.fn()} />);
  });

  it('shows the edit icon button', async () => {
    expect(await documentBody.findByTestId('EditIconButton')).toBeVisible();
    expect(documentBody.getByTestId('EditIcon')).toBeVisible();
  });

  it('matches the snapshot', async () => {
    const { baseElement } = documentBody;
    await waitForElementToBeRemoved(screen.getByRole('progressbar'));

    expect(baseElement).toMatchSnapshot();
  });
});
