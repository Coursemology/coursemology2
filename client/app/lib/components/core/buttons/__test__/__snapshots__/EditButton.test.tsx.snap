// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditButton /> matches the snapshot 1`] = `
<body>
  <div>
    <span
      aria-label="Edit"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-15u00r0-MuiButtonBase-root-MuiIconButton-root"
        data-testid="EditIconButton"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hgpioi-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </span>
    <div
      class="Toastify"
    />
  </div>
</body>
`;
